src/test/regress/sql/multi_tenant_isolation.sql

```diff
@@ -9,6 +9,18 @@ SELECT nextval('pg_catalog.pg_dist_placement_placementid_seq') AS last_placement
 \gset
 ALTER SEQUENCE pg_catalog.pg_dist_placement_placementid_seq RESTART 100000;
 
+CREATE OR REPLACE FUNCTION public.wait_for_resource_cleanup() RETURNS void AS $$
+DECLARE
+record_count integer;
+BEGIN
+    SET client_min_messages TO WARNING;
+    EXECUTE 'SELECT COUNT(*) FROM pg_catalog.pg_dist_cleanup' INTO record_count;
+    WHILE  record_count != 0 LOOP
+	 CALL pg_catalog.citus_cleanup_orphaned_resources();
+     EXECUTE 'SELECT COUNT(*) FROM pg_catalog.pg_dist_cleanup' INTO record_count;
+    END LOOP;
+    RESET client_min_messages;
+END$$ LANGUAGE plpgsql;
```

Comment:
You can set as part of the function definition. That way it rolls back correctly to whatever it was before ERROR after the call.
```suggestion
CREATE OR REPLACE FUNCTION public.wait_for_resource_cleanup() RETURNS void 
SET client_min_messages TO WARNING
AS $$
DECLARE
record_count integer;
BEGIN
    EXECUTE 'SELECT COUNT(*) FROM pg_catalog.pg_dist_cleanup' INTO record_count;
    WHILE  record_count != 0 LOOP
	 CALL pg_catalog.citus_cleanup_orphaned_resources();
     EXECUTE 'SELECT COUNT(*) FROM pg_catalog.pg_dist_cleanup' INTO record_count;
    END LOOP;
END$$ LANGUAGE plpgsql;
```